src/main/java/org/spongepowered/asm/mixin/transformer/MixinTargetContext.java

```diff
@@ -333,20 +334,72 @@ private void transformTypeNode(MethodNode method, Iterator<AbstractInsnNode> ite
     }
 
     /**
-     * Transforms class literals in the method being processed.
+     * Transforms class literals and method handle loads in the method being
+     * processed.
      * 
      * @param method Method being processed
      * @param iter Insn interator
      * @param ldcInsn Insn to transform
      */
     private void transformConstantNode(MethodNode method, Iterator<AbstractInsnNode> iter, LdcInsnNode ldcInsn) {
-        if (ldcInsn.cst instanceof Type) {
-            Type type = (Type)ldcInsn.cst;
-            String desc = this.transformSingleDescriptor(type);
+        ldcInsn.cst = this.transformConstant(method, iter, ldcInsn.cst);
+    }
+
+    /**
+     * Transforms a invoke dynamic instruction in the method being processed.
+     *
+     * @param method Method being processed
+     * @param iter Insn interator
+     * @param dynInsn Insn to transform
+     */
+    private void transformInvokeDynamicNode(MethodNode method, Iterator<AbstractInsnNode> iter, InvokeDynamicInsnNode dynInsn) {
+        dynInsn.desc = this.transformMethodDescriptor(dynInsn.desc);
+        dynInsn.bsm = this.transformHandle(method, iter, dynInsn.bsm);
+        for (int i = 0; i < dynInsn.bsmArgs.length; i++) {
+            dynInsn.bsmArgs[i] = this.transformConstant(method, iter, dynInsn.bsmArgs[i]);
+        }
+    }
+
+    /**
+     * Transforms a constant in the constant pool.
+     *
+     * @param method Method being processed
+     * @param iter Insn interator
+     * @param constant Consatnt pool entry
+     * @return Transformed constant
+     */
+    private Object transformConstant(MethodNode method, Iterator<AbstractInsnNode> iter, Object constant) {
+        if (constant instanceof Type) {
+            Type type = (Type)constant;
+            String desc = this.transformDescriptor(type);
             if (!type.toString().equals(desc)) {
-                ldcInsn.cst = Type.getType(desc);
+                return Type.getType(desc);
+            } else {
+                return constant;
             }
+        } else if (constant instanceof Handle) {
+            return this.transformHandle(method, iter, (Handle)constant);
+        }
+        return constant;
+    }
+
+    /**
+     * Transforms a method handle that is referenced from the method being
+     * processed.
+     *
+     * @param method Method being processed
+     * @param iter Insn interator
+     * @param handle Handle to transform
+     */
+    private Handle transformHandle(MethodNode method, Iterator<AbstractInsnNode> iter, Handle handle) {
+        this.targetClass.version = Constants.JAVA8_CLASSFILE_VERSION;
```

Comment:
did you know you could use Opcodes.V1_8 from asm instead of making mixins own constant?